### PR TITLE
CIP-116 | Add feedback from query layer standardisation workshop #2

### PR DESCRIPTION
**Pull request by Ryun1**
- small tidy/ corrections
- added "versioning" guidance
- added a changelog to be maintained 
- fleshed out rationale

### CODE REVIEW

File: CIP-0116/README.md [hunk 1]
```diff
@@ -247,21 +242,57 @@ Instead of providing all three variants of encoding, we base the schema on the o
...
+
 ## Rationale: how does this CIP achieve its goals?
 
-## Path to Active
+### Scope
 
-- [ ] Complete the specification
-- [ ] Provide an implementation of validating functions that uses this json-schema
-- [ ] Collect a list of cardano domain types implementations and negotiate transition to the specified formats with maintainers (if it makes sense and is possible)
+We keep a lean scope, limited to the data types within Cardano blocks. The rationale for this is that block data is by far the most useful for the majority of Cardano actors. There is also one nice benefit that the definitions can map directly from the provided CDDL file from ledger team.
```
Comment (klntsky) [inline CIP-0116/README.md]: > We keep a lean scope, limited to the data types within Cardano blocks

It does not sound like a lean scope. It's the most wide scope we could possibly have.

File: CIP-0116/README.md [hunk 2]
```diff
@@ -247,21 +242,57 @@ Instead of providing all three variants of encoding, we base the schema on the o
...
+
+Defining additional would likely add unneeded overhead to the maintenance of this standard.
```
Comment (klntsky) [inline CIP-0116/README.md]:
```suggestion
```

I think this gives no new information to the reader as there is no clue what these additional things could be

File: CIP-0116/README.md [hunk 3]
```diff
@@ -247,21 +242,57 @@ Instead of providing all three variants of encoding, we base the schema on the o
...
+
+### Strictness
+
+This CIP lays out strong conventions that future schema authors must follow. With a large set of design principles and conventions. The aim for these is to minimize the potential for unavoidable deltas between schemas.
```
Comment (klntsky) [inline CIP-0116/README.md]:
```suggestion
This CIP lays out strong conventions that future schema authors must follow, along with a large set of design principles and conventions. The aim is to minimize the potential for unavoidable deltas between schemas.
```

File: CIP-0116/README.md [hunk 4]
```diff
@@ -247,21 +242,57 @@ Instead of providing all three variants of encoding, we base the schema on the o
...
+
+By setting sometimes arbitrary conventions we hope to create a single possible interpretation from CBOR to JSON. This is beneficial as it allows the development of downstream standards, which rely on a canonical JSON representation.
```
Comment (klntsky) [inline CIP-0116/README.md]:
```suggestion
By setting sometimes arbitrary conventions we hope to create a single possible interpretation from CBOR to JSON, alleviating any ambiguity.
```
The rest was already stated in the motivation section.

File: CIP-0116/README.md [hunk 5]
```diff
@@ -247,21 +242,57 @@ Instead of providing all three variants of encoding, we base the schema on the o
...
+
+### JSON
+
+JSON was chosen as there is no viable alternative. The majority of Cardano's web tooling is built with Javascript where JSON is the primary object representation format.
+
+Furthermore, even across non-Javascript based stacks, JSON enjoys wide tooling support, this improves the potential for builders to adopt this standard.
+
+### Bech32 for addresses
+
+We choose to use Bech32 as the representation for Cardano addresses. When compared to the alternative of base64, Bech32 gives the advantages of an included checksum and a human readable prefix. These add additional safety to the standard.
```
Comment (klntsky) [inline CIP-0116/README.md]:
```suggestion
We choose to use Bech32 as the representation for Cardano addresses. When compared to the alternative of hexademical encoding, Bech32 gives the advantages of an included checksum and a human readable prefix.
```

File: CIP-0116/README.md [hunk 6]
```diff
@@ -247,21 +242,57 @@ Instead of providing all three variants of encoding, we base the schema on the o
...
+
+## Path to Active
 
 ### Acceptance Criteria
 
+- [ ] One future ledger era schema is added
+- [ ] This standard is implemented within three separate tools, libraries, etc.
```
Comment (klntsky) [inline CIP-0116/README.md:289]: Why three? Is this from some more general CIP guideline?

File: CIP-0116/README.md [hunk 7]
```diff
@@ -247,21 +242,57 @@ Instead of providing all three variants of encoding, we base the schema on the o
...
 
 ### Acceptance Criteria
 
+- [ ] One future ledger era schema is added
+- [ ] This standard is implemented within three separate tools, libraries, etc. 
+
 ### Implementation Plan
-<!-- A plan to meet those criteria. Or `N/A` if not applicable. -->
+
+- [ ] Complete the Babbage specification
```
Comment (klntsky) [inline CIP-0116/README.md]:
```suggestion
- [ ] Complete the Babbage specification
- [ ] Provide a test suite validating JSON fixtures for all the types against the schema 
```

File: CIP-0116/README.md (hunk 6 above)
Comment (Ryun1) [reply CIP-0116/README.md:289]: Feel free to change, I feel like it is a fair number, there is no official guidelines it is up to the author

described [here](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0001#acceptance-criteria)

File: CIP-0116/README.md (hunk 1 above)
Comment (Ryun1) [reply CIP-0116/README.md]: oops, I was being silly

Comment (Ryun1) [reply CIP-0116/README.md]: changed it

File: CIP-0116/README.md [hunk 8]
```diff
@@ -251,21 +246,56 @@ Instead of providing all three variants of encoding, we base the schema on the o
...
+### Scope
 
-- [x] Complete the specification for the current era
-- [x] Provide [an implementation](https://github.com/mlabs-haskell/cip-0116-tests) of validating functions that uses this json-schema
-- [ ] Collect a list of cardano domain types implementations and negotiate transition to the specified formats with maintainers (if it makes sense and is possible)
+We keep the scope of this standard to the data types within Cardano blocks. The rationale for this is that block data is by far the most useful for the majority of Cardano actors. There is also one nice benefit that the definitions can map directly from the provided CDDL file from ledger team.
+
+### Strictness
+
+This CIP lays out strong conventions that future schema authors must follow, along with a large set of design principles and conventions. The aim is to minimize the potential for unavoidable deltas between schemas.
```
Comment (klntsky) [inline CIP-0116/README.md]:
```suggestion
This CIP lays out strong conventions that future schema authors must follow, along with a large set of design principles. The aim is to minimize the potential for unavoidable deltas between schemas.
```

**Review by klntsky — CHANGES_REQUESTED** [review]

**Pull request merged**